packages/salesforcedx-vscode-apex/src/views/testOutlineProvider.ts

```diff
@@ -286,44 +233,38 @@ export class ApexTestGroupNode extends TestNode {
     this.skipping = 0;
     this.children.forEach(child => {
       if (child instanceof ApexTestNode) {
-        this.passing += child.outcome === PASS_RESULT ? 1 : 0;
-        this.failing += child.outcome === FAIL_RESULT ? 1 : 0;
-        this.skipping += child.outcome === SKIP_RESULT ? 1 : 0;
+        this.passing += child.outcome === 'Pass' ? 1 : 0;
+        this.failing += child.outcome === 'Fail' ? 1 : 0;
+        this.skipping += child.outcome === 'Skip' ? 1 : 0;
       }
     });
 
     if (this.passing + this.failing + this.skipping === this.children.length) {
-      if (this.failing !== 0) {
-        this.updateOutcome(FAIL_RESULT);
-      } else {
-        this.updateOutcome(PASS_RESULT);
-      }
+      this.updateOutcome(this.failing !== 0 ? 'Fail' : 'Pass');
     }
   }
 
-  public updateOutcome(outcome: string) {
+  public updateOutcome(outcome: TestOutcome) {
     super.updateOutcome(outcome);
-    if (outcome === PASS_RESULT) {
-      this.children.forEach(child => {
-        // Update all the children as well
-        child.updateOutcome(outcome);
-      });
+    if (outcome === 'Pass') {
+      // Update all the children as well
+      this.children.map(child => child.updateOutcome(outcome));
     }
   }
 }
 
 export class ApexTestNode extends TestNode {
   public errorMessage: string = '';
   public stackTrace: string = '';
-  public outcome = 'Not Run';
+  public readonly outcome: TestOutcome = 'Skip';
```

Comment:
set via `updateOutcome` instead of having to set it and then call updateOutcome on it.